plugins/inputs/sysstat/sysstat.go

```diff
@@ -199,6 +205,17 @@ func (s *Sysstat) collect() error {
 	return nil
 }
 
+func createTempFile() (string, error) {
+	tmpPath := path.Join(os.TempDir(), "tg-sysstat")
+	os.Mkdir(tmpPath, os.ModeDir)
```

Comment:
This directory needs to be removed, but since it isn't randomized we can't safely do this if there are multiple plugin instances running.  I suggest we use ioutil.TempDir and place the file inside.

Reply:
if Telegraf always reuses the same dir, there's no reason to ever remove it.

Reply:
We should always clean up our temporary files.

Reply:
You mean the directory, right? The files are getting removed as far as I can tell.

Reply:
Yes, all temporary files and directories should be cleaned up before Telegraf closes.